kinoml/datasets/kinomescan/pkis2.py

```diff
@@ -0,0 +1,109 @@
+from collections import defaultdict
+
+import pandas as pd
+
+from .utils import KINOMEScanMapper
+from ..core import BaseDatasetProvider
+from ...core.protein import AminoAcidSequence
+from ...core.ligand import Ligand
+from ...core.measurements import PercentageDisplacementMeasurement
+from ...core.conditions import AssayConditions
+from ...utils import datapath, defaultdictwithargs
+
+
+class PKIS2DatasetProvider:
+
+    """
+    Loads PKIS2 dataset as providd in `Progress towards a public chemogenomic set
+    for protein kinases and a call for contributions` (DOI: 10.1371/journal.pone.0181585)
+
+    It will build a dataframe where the SMILES-representation of ligands are the index
+    and the columns are the kinase names. To map between KINOMEscan kinase names and
+    actual sequences, helper object ``kinoml.datatasets.kinomescan.utils.KINOMEScanMapper``
+    is instantiated as a class attribute.
+
+    Parameters
+    ----------
+    featurizers : list of callables, optional=None
+        Callables that will modify the raw chemical data into other representations.
+
+    Attributes
+    ----------
+    kinases : lazy dict, str->AminoAcidSequence
+        Dict that will generate and cache AminoAcidSequence objects upon access,
+        with keys being any of the KINOMEScan kinase names
+    ligands : lazy dict, str->Ligand
+        Dict that will generate and cache Ligand objects upon access, with keys
+        being any of the available SMILES
+    available_kinases : list of str
+        All possible kinase names available in this dataset
+    available_ligands : list of str
+        All possible SMILES available in this dataset
+
+    Class attributes
+    ----------------
+    _RAW_DATASHEET : str
+        CSV file to load PKIS2 data from. If the file format is
+        different (columns, etc), sublass and reimplement ``self._read_dataframe``.
+
+    Examples
+    --------
+
+    >>> from kinoml.datasets.kinomescan.pkis2 import PKIS2DatasetProvider
+    >>> provider = PKIS2DatasetProvider()
+    >>> kin = provider.kinases["ABL2"]
+    >>> lig = provider.ligands[provider.available_ligands[0]]
+    >>> measurement = provider.measurements[kin, lig]
+    >>> print(f"% displacement for kinase={kin.header} and ligand={lig.to_smiles()} is {measurement}"
+
+    """
+
+    _RAW_SPREADSHEET = datapath("kinomescan/journal.pone.0181585.s004.csv")
+    _kinase_name_mapper = KINOMEScanMapper()
+
+    ASSAY_CONDITIONS = AssayConditions(pH=7.0)
```

Comment:
Setting these fields at the class level is dangerous---these could be changed and lead to incorrect results!

Reply:
My idea with this approach is that this class _only_ offers support for the PKIS2 dataset, as published. If you want to use it for another dataset that relies on the Kinomescan platform, then you should subclass it. This would also address the potential issue of having slightly different CSV columns or other variations of the input.

The other approach is providing these defaults as keyword arguments. As you wish!